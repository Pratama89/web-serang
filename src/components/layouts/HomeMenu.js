import Image from "next/image";
import MenuItem from "../menu/MenuItem";
import SectionHeaders from "./SectionHeader";
import Link from "next/link";

export default function HomeMenu() {
  return (
    <section className="mt-20 pt-24" id="product">
      <div className="text-center">
        <SectionHeaders
          subHeader="Jasa dan Layanan Kami"
          mainHeader="Kami mengerjakan pengeleman Kain dan Busa dengan kualitas terbaik."
        />
      </div>
      <div className="grid grid-cols-2 gap-6 mt-8 w-full xs:grid-cols-1 xs:p-5 xs:gap-4 px-20">
        <div className="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] ">
          <Image
            src={"/img/proses-laminating2.jpg"}
            alt="kain"
            width={200}
            height={200}
            className="h-56 w-56 bg-cover rounded-lg delay-[100ms] duration-[300ms] hover:scale-[1.1] xs:h-36 xs:w-36"
          />
          <div className="xs:py-2">
            <h1 className="text-base font-medium mx-4">Laminating Kain PE dan Busa</h1>
            <p className="text-xs my-2 mx-4">
              Busa lapis dengan material terbaik bisa untuk exterior produk,
              lapisan kain BH, aksesori, dan lain sebagainya.
            </p>
            <Link href="https://wa.link/h7ywr2">
              <button className="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 xs:py-1 xs:px-3 focus:outline-none hover:bg-red-600 rounded  text-center">
                Hubungi Admin
              </button>
            </Link>
          </div>
        </div>
        <div className="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] ">
          <Image
            src={"/img/insole sepatu.png"}
            alt="kain"
            width={200}
            height={200}
            className="h-56 w-56 bg-cover rounded-lg delay-[100ms] duration-[600ms] hover:scale-[1.1] xs:h-36 xs:w-36"
          />
          <div className="xs:py-2">
            <h1 className="text-base font-medium ">
              Laminating Insole Sepatu
            </h1>
            <p className="text-xs my-2 mx-2">
            Cup Insole merupakan komponen sepatu yang penting adanya sebagai alas untuk menambah kenyamanan pengguna.
            </p>
            <Link href="https://wa.link/h7ywr2">
              <button className="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 xs:py-1 xs:px-3 focus:outline-none hover:bg-red-600 rounded  text-center">
                Hubungi Admin
              </button>
            </Link>
          </div>
        </div>
        <div className="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] ">
          <Image
            src="/img/busa-kasur.jpg"
            alt="kain"
            width={200}
            height={200}
            className="h-56 w-56 bg-cover rounded-lg delay-[100ms] duration-[600ms] hover:scale-[1.1] xs:h-36 xs:w-36"
          />
          <div className="xs:py-2">
            <h1 className="text-base font-medium ">
              Laminating Busa dan Kain Kasur
            </h1>
            <p className="text-xs my-2 mx-2">
              Kami menerima juga laminating busa dan kain kasur
            </p>
            <Link href="https://wa.link/h7ywr2">
              <button className="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 xs:py-1 xs:px-3 focus:outline-none hover:bg-red-600 rounded  text-center">
                Hubungi Admin
              </button>
            </Link>
          </div>
        </div>
        <div className="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] ">
          <Image
            src="/img/admin2.jpg"
            alt="kain"
            width={200}
            height={200}
            className="h-56 w-56 bg-cover rounded-lg delay-[100ms] duration-[200ms] hover:scale-[1.1] xs:h-36 xs:w-36"
          />
          <div className="xs:py-2">
            <h3 className="text-base font-medium">Pelayanan Yang Ramah</h3>
            <p className="text-xs my-2 mx-2">
              Data masuk dan keluar tercatat rapih oleh admin kami, sehingga
              proses hasil laminating bisa berjalan lancar.
            </p>
            <Link href="https://wa.link/h7ywr2">
              <button className="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 xs:py-1 xs:px-3 focus:outline-none hover:bg-red-600 rounded  text-center">
                Hubungi Admin
              </button>
            </Link>
          </div>
        </div>
      </div>
    </section>
  );
}
