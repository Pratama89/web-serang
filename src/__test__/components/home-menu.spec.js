import HomeMenu from "@/components/layouts/HomeMenu";
import { render, screen } from "@testing-library/react";
import { describe } from "node:test";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/homemenu",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    };
  },
}));

describe("HomeMenu", () => {
  it("should render", () => {
    const page = render(<HomeMenu />);
    // expect(screen.getByTestId("title").textContent).toBe("Header");
    expect(page).toMatchSnapshot();
  });
});
