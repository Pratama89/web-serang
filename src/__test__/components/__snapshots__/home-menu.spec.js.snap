// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="mt-40"
        id="menu"
      >
        <div
          class="text-center"
        >
          <div>
            <h3
              class="capitalize text-primary font-semibold text-3xl"
            >
              Produk dan Layanan Kami
            </h3>
            <h2
              class="text-base text-gray-600 font-normal italic mt-2 md:text-sm lg:text-lg"
            >
              Kami mengerjakan pengeleman Kain dan Busa dengan kualitas terbaik.
            </h2>
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-6 mt-8 w-full xs:grid-cols-1"
        >
          <div
            class="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] "
          >
            <img
              alt="kain"
              class="bg-cover rounded-lg delay-[100ms] duration-[300ms] hover:scale-[1.1]"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=%2Fimg%2Fproses-laminating2.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fimg%2Fproses-laminating2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproses-laminating2.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="200"
            />
            <div
              class=""
            >
              <h3
                class="text-base font-medium mx-4"
              >
                Laminating Busa
              </h3>
              <p
                class="text-xs my-2 mx-4"
              >
                Busa lapis dengan material terbaik bisa untuk exterior produk, lapisan kain BH, aksesori, dan lain sebagainya.
              </p>
              <a
                href="https://wa.link/h7ywr2"
              >
                <button
                  class="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 focus:outline-none hover:bg-red-600 rounded  text-center"
                >
                  Hubungi Admin
                </button>
              </a>
            </div>
          </div>
          <div
            class="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] "
          >
            <img
              alt="kain"
              class="bg-cover rounded-lg delay-[100ms] duration-[600ms] hover:scale-[1.1]"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=%2Fimg%2Fproses-finishing2.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fimg%2Fproses-finishing2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproses-finishing2.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="200"
            />
            <div
              class=""
            >
              <h3
                class="text-base font-medium "
              >
                Proses Finishing Laminating
              </h3>
              <p
                class="text-xs my-2 mx-2"
              >
                Bahan yang sudah  selesai dilaminating akan segera di bungkus dengan plastik secara aman.
              </p>
              <a
                href="https://wa.link/h7ywr2"
              >
                <button
                  class="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 focus:outline-none hover:bg-red-600 rounded  text-center"
                >
                  Hubungi Admin
                </button>
              </a>
            </div>
          </div>
          <div
            class="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] "
          >
            <img
              alt="kain"
              class="bg-cover rounded-lg delay-[100ms] duration-[600ms] hover:scale-[1.1]"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=%2Fimg%2Fproses-potong-bis2.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fimg%2Fproses-potong-bis2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproses-potong-bis2.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="200"
            />
            <div
              class=""
            >
              <h3
                class="text-base font-medium "
              >
                Proses Persiapan Potong Kain Bis
              </h3>
              <p
                class="text-xs my-2 mx-2"
              >
                Kain sisa potongan busa biasanya di potong untuk kain bis.
              </p>
              <a
                href="https://wa.link/h7ywr2"
              >
                <button
                  class="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 focus:outline-none hover:bg-red-600 rounded  text-center"
                >
                  Hubungi Admin
                </button>
              </a>
            </div>
          </div>
          <div
            class="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] "
          >
            <img
              alt="kain"
              class="bg-cover rounded-lg delay-[100ms] duration-[200ms] hover:scale-[1.1]"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=%2Fimg%2Fadmin2.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fimg%2Fadmin2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fadmin2.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="200"
            />
            <div
              class=""
            >
              <h3
                class="text-base font-medium"
              >
                Pelayanan Yang Ramah
              </h3>
              <p
                class="text-xs my-2 mx-2"
              >
                Data masuk dan keluar tercatat rapih oleh admin kami, sehingga proses hasil laminating bisa berjalan lancar.
              </p>
              <a
                href="https://wa.link/h7ywr2"
              >
                <button
                  class="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 focus:outline-none hover:bg-red-600 rounded  text-center"
                >
                  Hubungi Admin
                </button>
              </a>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="mt-40"
      id="menu"
    >
      <div
        class="text-center"
      >
        <div>
          <h3
            class="capitalize text-primary font-semibold text-3xl"
          >
            Produk dan Layanan Kami
          </h3>
          <h2
            class="text-base text-gray-600 font-normal italic mt-2 md:text-sm lg:text-lg"
          >
            Kami mengerjakan pengeleman Kain dan Busa dengan kualitas terbaik.
          </h2>
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-6 mt-8 w-full xs:grid-cols-1"
      >
        <div
          class="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] "
        >
          <img
            alt="kain"
            class="bg-cover rounded-lg delay-[100ms] duration-[300ms] hover:scale-[1.1]"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fimg%2Fproses-laminating2.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg%2Fproses-laminating2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproses-laminating2.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
          <div
            class=""
          >
            <h3
              class="text-base font-medium mx-4"
            >
              Laminating Busa
            </h3>
            <p
              class="text-xs my-2 mx-4"
            >
              Busa lapis dengan material terbaik bisa untuk exterior produk, lapisan kain BH, aksesori, dan lain sebagainya.
            </p>
            <a
              href="https://wa.link/h7ywr2"
            >
              <button
                class="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 focus:outline-none hover:bg-red-600 rounded  text-center"
              >
                Hubungi Admin
              </button>
            </a>
          </div>
        </div>
        <div
          class="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] "
        >
          <img
            alt="kain"
            class="bg-cover rounded-lg delay-[100ms] duration-[600ms] hover:scale-[1.1]"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fimg%2Fproses-finishing2.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg%2Fproses-finishing2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproses-finishing2.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
          <div
            class=""
          >
            <h3
              class="text-base font-medium "
            >
              Proses Finishing Laminating
            </h3>
            <p
              class="text-xs my-2 mx-2"
            >
              Bahan yang sudah  selesai dilaminating akan segera di bungkus dengan plastik secara aman.
            </p>
            <a
              href="https://wa.link/h7ywr2"
            >
              <button
                class="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 focus:outline-none hover:bg-red-600 rounded  text-center"
              >
                Hubungi Admin
              </button>
            </a>
          </div>
        </div>
        <div
          class="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] "
        >
          <img
            alt="kain"
            class="bg-cover rounded-lg delay-[100ms] duration-[600ms] hover:scale-[1.1]"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fimg%2Fproses-potong-bis2.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg%2Fproses-potong-bis2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproses-potong-bis2.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
          <div
            class=""
          >
            <h3
              class="text-base font-medium "
            >
              Proses Persiapan Potong Kain Bis
            </h3>
            <p
              class="text-xs my-2 mx-2"
            >
              Kain sisa potongan busa biasanya di potong untuk kain bis.
            </p>
            <a
              href="https://wa.link/h7ywr2"
            >
              <button
                class="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 focus:outline-none hover:bg-red-600 rounded  text-center"
              >
                Hubungi Admin
              </button>
            </a>
          </div>
        </div>
        <div
          class="bg-gray-100 border shadow-lg  rounded-lg text-center flex items-center hover:bg-white delay-[100ms] duration-[200ms] hover:scale-[1.1] "
        >
          <img
            alt="kain"
            class="bg-cover rounded-lg delay-[100ms] duration-[200ms] hover:scale-[1.1]"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fimg%2Fadmin2.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg%2Fadmin2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fadmin2.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
          <div
            class=""
          >
            <h3
              class="text-base font-medium"
            >
              Pelayanan Yang Ramah
            </h3>
            <p
              class="text-xs my-2 mx-2"
            >
              Data masuk dan keluar tercatat rapih oleh admin kami, sehingga proses hasil laminating bisa berjalan lancar.
            </p>
            <a
              href="https://wa.link/h7ywr2"
            >
              <button
                class="mx-2 mt-4 text-white bg-red-500 border-0 py-2 px-6 focus:outline-none hover:bg-red-600 rounded  text-center"
              >
                Hubungi Admin
              </button>
            </a>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
